dvc/remote/base.py

```diff
@@ -899,27 +906,35 @@ def _checkout_dir(
 
             entry_checksum_info = {self.PARAM_CHECKSUM: entry_checksum}
             if relink or self.changed(entry_info, entry_checksum_info):
+                _modified = True
                 self.safe_remove(entry_info, force=force)
                 self.link(entry_cache_info, entry_info)
                 self.state.save(entry_info, entry_checksum)
             if progress_callback:
                 progress_callback(str(entry_info))
 
-        self._remove_redundant_files(path_info, dir_info, force)
+        _modified = _modified or self._remove_redundant_files(
+            path_info, dir_info, force
+        )
 
         self.state.save_link(path_info)
         self.state.save(path_info, checksum)
 
+        if _added or _modified:
+            return not _added and _modified
+
```

Comment:
Shouldn't we return something when `_added=False` and `_modified=False`?

Reply:
I understand it's confusing, and I'll probably refactor it to return some kind of enum status. I'll leave it as it is for now, till I know more requirements (or, if this suffice). Thanks for the review!